docs/current/guides/421437-work-with-host-filesystem.md

```diff
@@ -212,6 +212,28 @@ The following example creates a file in a container's `/tmp` directory and then
 </TabItem>
 </Tabs>
 
+## Important notes
+
+Using the host filesystem in your Dagger pipeline is convenient, but there are some important considerations to keep in mind:
+
+- If a file loaded from the host changes even slightly (including minor changes such as a timestamp change with the file contents left unmodified), then the Dagger cache will be invalidated. An extremely common source of invalidations occurs when loading the `.git` directory from the host filesystem, as that directory is constantly changing.
+
+  :::tip
+  To maximize cache re-use, it's important to use the include/exclude options for local directories to only include the files/directories needed for the pipeline. Excluding the `.git` directory is highly advisable.
+  :::
+
+- The host directory is loaded into the Dagger Engine; it's not a "bind mount". This means that any change you make to the loaded directory in your Dagger pipeline will not result in a change to the directory on the host.
+
+  :::warning
+  If you want the changes made to a loaded local directory inside a Dagger pipeline to be reflected on the host, it needs to be explictly exported to the host. However, this should be approached with great caution, since a mistake could result in files/directories being accidentally overwritten with undesired contents or unsaved work being deleted.
```

Comment:
```suggestion
  If you want the changes made to a loaded local directory inside a Dagger pipeline to be reflected on the host, it needs to be explictly exported to the host. However, this should be approached with caution, since any overlap in the files being exported with the files on the host will result in the host files being overwritten.
```

This is an important note but exporting is a pretty common use case and while caution is advised, I think we can tone down a tad here.